aten/src/ATen/native/cuda/ForeachTensorAddList.cu

```diff
@@ -0,0 +1,47 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/ForeachUtils.h>
+#include <ATen/native/cuda/ForeachFunctors.cuh>
+
+namespace at { namespace native {
+
+std::vector<Tensor> foreach_tensor_add_list_kernel_cuda(TensorList tensors1, TensorList tensors2) {
+    verify_list(tensors1, tensors2);
+
+    if (!check_fast_route(tensors1, tensors2)) {
+        return at::native::foreach_tensor_add_list_kernel_slow(tensors1, tensors2);
+    }
+
+    std::vector<std::vector<at::Tensor>> tensor_lists; 
```

Comment:
(no action required) We know exactly how many lists there needs to be, so it would be nice if we could avoid the dynamic allocation here. This would require us to revisit the first argument to `multi_tensor_apply`.